xmtp_id/src/scw_verifier/mod.rs

```diff
@@ -87,32 +101,26 @@ impl MultiSmartContractSignatureVerifier {
         Self { verifiers }
     }
 
-    pub fn new_from_file(path: impl AsRef<Path>) -> Self {
-        let path = path.as_ref();
-
-        let file_str;
-        let json = if path.exists() {
-            file_str = fs::read_to_string(path).unwrap_or_else(|_| panic!("{path:?} is missing"));
-            &file_str
-        } else {
-            DEFAULT_CHAIN_URLS
-        };
-
-        let json: HashMap<String, String> =
-            serde_json::from_str(json).unwrap_or_else(|_| panic!("{path:?} is malformatted"));
+    pub fn new_from_file(path: impl AsRef<Path>) -> Result<Self, io::Error> {
+        let json = fs::read_to_string(path.as_ref())?;
+        let urls: HashMap<String, Url> = serde_json::from_str(&json).map_err(|err| {
+            io::Error::new(
+                io::ErrorKind::InvalidData,
+                format!("Unable to deserialize json: {err:?}"),
+            )
+        })?;
 
-        let urls = json
-            .into_iter()
-            .map(|(id, url)| {
-                (
-                    id,
-                    Url::from_str(&url)
-                        .unwrap_or_else(|_| panic!("unable to parse url in {path:?} ({url})")),
-                )
-            })
-            .collect();
+        Ok(Self::new(urls))
+    }
 
-        Self::new(urls)
+    /// Upgrade the default urls to paid/private urls if the env vars are present.
+    pub fn upgrade(mut self) -> Self {
+        self.verifiers.iter_mut().for_each(|(id, verif)| {
+            if let Ok(url) = env::var(format!("CHAIN_RPC_{id}")) {
```

Comment:
It would be helpful to log if any are missing, so we can't forget to set an environment variable and silently fall back to the defaults.

Reply:
Good point